motor-control/firmware/motor_control_hardware.c

```diff
@@ -86,7 +86,12 @@ uint32_t motor_hardware_encoder_pulse_count(void* enc_htim) {
 }
 
 void motor_hardware_reset_encoder_count(void* enc_htim) {
-    if (enc_htim != NULL) {
-        __HAL_TIM_SET_COUNTER((TIM_HandleTypeDef*)enc_htim, 0);
+    /* Offset the zero position by 10 pulses to eliminate any noise on the system.
+     */
+    bool overflow_flag = __HAL_TIM_GET_FLAG((TIM_HandleTypeDef*)enc_htim, TIM_FLAG_UPDATE);
+    while(overflow_flag){
```

Comment:
Can you explain what's going on here a little more. When an overflow happens, the flag will be 1 and then we immediately update again? Is this handling an edge case of some sort?

Reply:
This is the reset that gets called after home; this code resets the overflow flag in case it happened to be set at that time. The loop I guess makes sure it's cleared. I don't love this, and I think I'll remove it.